Lib/pathlib.py

```diff
@@ -822,10 +829,6 @@ def match(self, path_pattern):
                 return False
         return True
 
-# Can't subclass os.PathLike from PurePath and keep the constructor
-# optimizations in PurePath._parse_args().
-os.PathLike.register(PurePath)
```

Comment:
This was added by @brettcannon when adding support for path-like to pathlib.

Reply:
Yep. I'm guessing that changing the metaclass somehow broke `_parse_args()` or made it slower, but in my testing I couldn't see any issues.

If inheriting from `os.PathLike` is an issue, I assume inheriting from `abc.ABC` later on will cause the same issue. Hoping Brett can shed some light :-)

Reply:
Best guess (since I don't remember) is at the time the code dynamically chose what to return, and so subclassing didn't do the appropriately thing (or something).

Wouldn't be surprised if it is no longer a concern (whatever it was 😅).

Reply:
Ah, right. Think 2219187cab6bca009c42b63b2f4c30b5b10c916d might have addressed that!

Reply:
Tentatively resolving this thread, as I can't detect any issues